Add `$content` parameter to `consul::watch`

joshuaspence opened the pull request:
Adds a `$content` parameter to `consul::watch` to allow more flexibility when creating a template.

craigwatson reviewed (changes requested):
Hi, thanks for the submission - we'd definitely like to get your fork back in sync, this approach does seem a lot cleaner than individual parameters for config values.

Can you rebase your changes against our current master?

Commit:
Add `$content` parameter to `consul::watch`

Adds a `$content` parameter to `consul::watch` to allow more flexibility when creating a template.
